### PR TITLE
DOC: Better docs for `jnp.left_shift`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/ufuncs.py [hunk 1]
```diff
@@ -1229,14 +1229,63 @@ def _bitwise_xor(x: ArrayLike, y: ArrayLike, /) -> Array:
...
+    x2: The amount of bits to shift each element in ``x1`` to the left, only accepts
+      integer subtypes. ``x1`` and ``x2`` must either have same shape or be broadcast
+      compatible.
+
+  Returns:
+    An array containing the left shifted elements of ``x1`` by the amount specified
+    in ``x2``, with the same shape as the broadcasted shape of ``x1`` and ``x2``.
+
+  Note:
+    Left shifting ``x1`` by ``x2`` is equivalent to ``x1 * (2**x2)``.
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Thanks! Modified the Note to `"Left shifting ``x1`` by ``x2`` is equivalent to ``x1 * (2**x2)`` within the bounds of the dtypes involved."`

File: jax/_src/numpy/ufuncs.py [hunk 2]
```diff
@@ -1229,14 +1229,63 @@ def _bitwise_xor(x: ArrayLike, y: ArrayLike, /) -> Array:
...
-@implements(np.left_shift, module='numpy')
+
 @partial(jit, inline=True)
 def left_shift(x: ArrayLike, y: ArrayLike, /) -> Array:
+  r"""Shift bits of ``x1`` to left by the amount specified in ``x2``, element-wise.
+
+  JAX implementation of :obj:`numpy.left_shift`.
+
+  Args:
+    x1: Input array, only accepts unsigned integer subtypes
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Modified it as `"x1: Input array, only accepts unsigned and signed integer subtypes."`

File: jax/_src/numpy/ufuncs.py [hunk 3]
```diff
@@ -1229,14 +1229,63 @@ def _bitwise_xor(x: ArrayLike, y: ArrayLike, /) -> Array:
...
+    >>> print_binary(result1)
+    ['0b1000', '0b10000', '0b100000', '0b1000000']
+  """
   return lax.shift_left(*promote_args_numeric("left_shift", x, y))
 
-@implements(getattr(np, "bitwise_left_shift", np.left_shift), module='numpy')
+
 @partial(jit, inline=True)
 def bitwise_left_shift(x: ArrayLike, y: ArrayLike, /) -> Array:
+  """Alias of :func:`jax.numpy.left_shift`"""
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done

File: jax/_src/numpy/ufuncs.py [hunk 4]
```diff
@@ -1229,14 +1229,64 @@ def _bitwise_xor(x: ArrayLike, y: ArrayLike, /) -> Array:
...
+  Returns:
+    An array containing the left shifted elements of ``x1`` by the amount specified
+    in ``x2``, with the same shape as the broadcasted shape of ``x1`` and ``x2``.
+
+  Note:
+    Left shifting ``x1`` by ``x2`` is equivalent to ``x1 * (2**x2)`` within the
+    bounds of the dtypes involved.
+
+  See also:
+    - :func:`jax.numpy.right_shift`: and :func:`jax.numpy.bitwise_left_shift`: Shifts
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Done.

File: jax/_src/numpy/ufuncs.py [hunk 5]
```diff
@@ -1229,14 +1229,64 @@ def _bitwise_xor(x: ArrayLike, y: ArrayLike, /) -> Array:
...
-@implements(np.left_shift, module='numpy')
+
 @partial(jit, inline=True)
 def left_shift(x: ArrayLike, y: ArrayLike, /) -> Array:
+  r"""Shift bits of ``x1`` to left by the amount specified in ``x2``, element-wise.
+
+  JAX implementation of :obj:`numpy.left_shift`.
+
+  Args:
+    x1: Input array, only accepts unsigned and signed integer subtypes.
```
[the remark this answers is not in the document]
Comment (rajasekharporeddy) [reply jax/_src/numpy/ufuncs.py]: Thanks! Done

**Review by jakevdp — APPROVED** [review]
Thanks!

**Pull request merged**